src/factory/mod.rs

```diff
@@ -468,7 +485,26 @@ impl Factory {
         Source::with_object(object)
     }
 
-    /// Update the geometry of `DynamicMesh`.
+    /// Update the vertex positions of the geometry of a `DynamicMesh`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if any of the indices exceed the number of vertices in the
+    /// mesh geometry.
+    pub fn update_mesh_vertices<I: Iterator<Item = (usize, mint::Point3<f32>)>>(
```

Comment:
I think we should just provide a mapping API for dynamic meshes, e.g.
```rust
{
    let mut mapping = factory.map_vertices(&mesh);
    for i in 0.. 5 {
         mapping[i] = MyVertex {..};
    }
}
```
This mapping object would implement `DerefMut` to our vertex format and hold `factory` by a mutable reference, then issue an actual update message upon destruction (a message would make the renderer to copy the data from upload -> data buffer upon the first use or something).

Reply:
This would be a much simpler approach. The downside is having to expose out vertex format.